docs/analyzers/Using Additional Files.md

```diff
@@ -107,38 +107,38 @@ public class CheckTermsAnalyzer : DiagnosticAnalyzer
             DiagnosticSeverity.Error,
             isEnabledByDefault: true);
 
-    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
     public override void Initialize(AnalysisContext context)
     {
-        context.RegisterCompilationStartAction(compilationStartContext =>
+        context.RegisterCompilationStartAction(context =>
         {
             // Find the file with the invalid terms.
-            ImmutableArray<AdditionalText> additionalFiles = compilationStartContext.Options.AdditionalFiles;
+            ImmutableArray<AdditionalText> additionalFiles = context.Options.AdditionalFiles;
             AdditionalText termsFile = additionalFiles.FirstOrDefault(file => Path.GetFileName(file.Path).Equals("Terms.txt"));
 
             if (termsFile != null)
             {
                 HashSet<string> terms = new HashSet<string>();
```

Comment:
@sharwell Can this cause some concurrency issues?

Reply:
It's fine as long as the writes to `terms` only occur within the compilation start callback. Other callbacks do not occur until this callback completes.